eth/downloader/stagedsync_stage_hashcheck.go

```diff
@@ -1,41 +1,65 @@
 package downloader
 
 import (
+	"bufio"
+	"bytes"
+	"container/heap"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"sort"
 
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/common/dbutils"
+	"github.com/ledgerwatch/turbo-geth/core"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/ethdb"
 	"github.com/ledgerwatch/turbo-geth/log"
 	"github.com/ledgerwatch/turbo-geth/trie"
+
 	"github.com/pkg/errors"
+	"github.com/ugorji/go/codec"
 )
 
-func (d *Downloader) spawnCheckFinalHashStage(syncHeadNumber uint64) error {
-	hashProgress, err := GetStageProgress(d.stateDB, HashCheck)
+var cbor codec.CborHandle
+
+func spawnCheckFinalHashStage(stateDB ethdb.Database, syncHeadNumber uint64, datadir string) error {
+	hashProgress, err := GetStageProgress(stateDB, HashCheck)
 	if err != nil {
 		return err
 	}
 
-	//REMOVE THE FOLLOWING LINE WHEN PLAIN => HASHED TRANSFORMATION IS READY
```

Comment:
Can you keep that in for now, please? this is to avoid a very expensive step for our first user :)